jsonalchemy/reader.py

```diff
@@ -422,8 +422,11 @@ def set_default_value(field, schema):
                 self._json.__setitem__(field_name, value, extend=False,
                                        exclude=['decorators', 'extensions'])
             else:
-                old_value = self._json.__getitem__(
-                    field_name, exclude=['decorators', 'extensions'])
+                try:
+                    old_value = self._json.__getitem__(
+                        field_name, exclude=['decorators', 'extensions'])
+                except KeyError:
+                    old_value = value
```

Comment:
I remember what you told me, but it would be nice if you also share it with the rest ;-)

And, indeed, an example or test case would be nice also.